packages/amplify-gen2-codegen/src/backend/synthesizer.ts

```diff
@@ -225,7 +239,147 @@ export class BackendSynthesizer {
       }
     }
 
-    return nativeUserPoolClientExpressionStatement;
+    return userPoolClientDeclaration;
+  }
+
+  private createPropertyAccessChain(identifiers: string[]): ts.Expression {
+    return identifiers
+      .slice(1)
+      .reduce<ts.Expression>(
+        (acc, curr) => factory.createPropertyAccessExpression(acc, factory.createIdentifier(curr)),
+        factory.createIdentifier(identifiers[0]),
+      );
+  }
+
+  private createProviderSetupCode(): ts.Statement[] {
+    // Create const providerSetupResult = (backend.auth.stack.node.children.find(child => child.node.id === "amplifyAuth") as any).providerSetupResult;
+    const providerSetupDeclaration = factory.createVariableStatement(
+      undefined,
+      factory.createVariableDeclarationList(
+        [
+          factory.createVariableDeclaration(
+            factory.createIdentifier('providerSetupResult'),
+            undefined,
+            undefined,
+            factory.createPropertyAccessExpression(
+              factory.createParenthesizedExpression(
+                factory.createAsExpression(
+                  factory.createCallExpression(
+                    factory.createPropertyAccessExpression(
+                      this.createPropertyAccessChain(['backend', 'auth', 'stack', 'node', 'children']),
+                      factory.createIdentifier('find'),
+                    ),
+                    undefined,
+                    [
+                      factory.createArrowFunction(
+                        undefined,
+                        undefined,
+                        [factory.createParameterDeclaration(undefined, undefined, factory.createIdentifier('child'))],
+                        undefined,
+                        factory.createToken(ts.SyntaxKind.EqualsGreaterThanToken),
+                        factory.createBinaryExpression(
+                          this.createPropertyAccessChain(['child', 'node', 'id']),
+                          factory.createToken(ts.SyntaxKind.EqualsEqualsEqualsToken),
+                          factory.createStringLiteral('amplifyAuth'),
+                        ),
+                      ),
+                    ],
+                  ),
+                  factory.createKeywordTypeNode(ts.SyntaxKind.AnyKeyword),
+                ),
+              ),
+              factory.createIdentifier('providerSetupResult'),
+            ),
+          ),
+        ],
+        ts.NodeFlags.Const,
+      ),
+    );
+
+    // Create Object.keys(providerSetupResult).forEach(...)
+    const forEachStatement = factory.createExpressionStatement(
```

Comment:
can we move the for loop contents into a separate method to reduce the size of this method?